geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/key/ScanExecutor.java

```diff
@@ -103,7 +104,7 @@ public RedisResponse executeCommand(Command command, ExecutionHandlerContext con
         scan(getDataRegion(context).keySet(), matchPattern, count, cursor);
     context.setScanCursor(scanResult.getLeft());
 
-    return RedisResponse.scan(scanResult.getLeft(), scanResult.getRight());
+    return RedisResponse.scan(scanResult.getLeft().intValue(), scanResult.getRight());
   }
 
   private Pair<BigInteger, List<Object>> scan(Collection<RedisKey> list,
```

Comment:
why does this BigInteger flavor still exist

Reply:
I didn't want to have this PR get too big, so I haven't changed the existing implementations for SCAN or SSCAN. They're currently not supported commands, so I've left that work for whoever eventually implements them in the future.